File: exocheat v2/exocheat/events/UpdateEvent.java
package club.shmoke.main.exocheat.events;

import org.bukkit.event.Cancellable;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

/**
 * @author Christian
 */
public class UpdateEvent extends Event {

    private static HandlerList handlerList = new HandlerList();

    public HandlerList getHandlers() {
        return handlerList;
    }

    public static HandlerList getHandlerList() {
        return handlerList;
    }

}

File: fbiclient v2/club/shmoke/main/events/UpdateEvent.kt
package club.shmoke.main.events

import club.shmoke.api.event.Event
import net.minecraft.client.entity.EntityPlayerSP

/**
 * @author Christian
 */
class UpdateEvent : Event {

    var type: Type? = Type.POST
    var y: Double = 0.toDouble()
    var yaw: Float = 0.toFloat()
    var pitch: Float = 0.toFloat()

    constructor(y: Double, yaw: Float, pitch: Float) {
        type = Type.PRE
        this.y = y
        this.yaw = yaw
        this.pitch = pitch
    }

    constructor() {
        type = Type.POST
    }

    enum class Type {
        PRE, POST
    }
}
